Sources/Datadog/Tracing/Propagation/OpenTelemetry/OpenTelemetryHTTPHeadersWriter.swift

```diff
@@ -0,0 +1,135 @@
+    public init(
+        samplingRate: Float = 20,
+        openTelemetryHeaderType: OpenTelemetryHeaderType
+    ) {
+        self.sampler = Sampler(samplingRate: samplingRate)
+        self.openTelemetryHeaderType = openTelemetryHeaderType
+    }
+
+    /// Creates a `OpenTelemetryHTTPHeadersWriter` to inject traces propagation headers
+    /// to network request.
+    ///
+    /// - Parameter sampler: Tracing sampler responsible for randomizing the sample.
+    /// - Parameter openTelemetryHeaderType: Determines the type of telemetry header type used by the writer.
+    internal init(
+        sampler: Sampler,
+        openTelemetryHeaderType: OpenTelemetryHeaderType
+    ) {
+        self.sampler = sampler
+        self.openTelemetryHeaderType = openTelemetryHeaderType
+    }
+
+    public func inject(spanContext: OTSpanContext) {
+        guard let spanContext = spanContext.dd else {
+            return
+        }
+
+        let samplingPriority = sampler.sample()
+
+        switch openTelemetryHeaderType {
+        case .multiple:
+            tracePropagationHTTPHeaders = [
+                OpenTelemetryHTTPHeaders.Multiple.sampledField: samplingPriority ? Constants.sampledValue : Constants.unsampledValue
+            ]
+
+            if samplingPriority {
+                tracePropagationHTTPHeaders[OpenTelemetryHTTPHeaders.Multiple.traceIDField] = spanContext.traceID.toHexadecimalString
+                tracePropagationHTTPHeaders[OpenTelemetryHTTPHeaders.Multiple.spanIDField] = spanContext.spanID.toHexadecimalString
+                if let parentSpanID = spanContext.parentSpanID {
+                    tracePropagationHTTPHeaders[OpenTelemetryHTTPHeaders.Multiple.parentSpanIDField] = parentSpanID.toHexadecimalString
+                }
+            }
+        case .single:
+            if samplingPriority {
+                let parentSpanIdHexadecimalString: String?
+                if let parentSpanID = spanContext.parentSpanID {
+                    parentSpanIdHexadecimalString = parentSpanID.toHexadecimalString
+                } else {
+                    parentSpanIdHexadecimalString = nil
+                }
+                tracePropagationHTTPHeaders[OpenTelemetryHTTPHeaders.Single.b3Field] = [
+                    spanContext.traceID.toHexadecimalString,
+                    spanContext.spanID.toHexadecimalString,
+                    Constants.sampledValue,
+                    parentSpanIdHexadecimalString
+                ]
+                .compactMap { $0 }
+                .joined(separator: Constants.b3Separator)
+            } else {
+                tracePropagationHTTPHeaders[OpenTelemetryHTTPHeaders.Single.b3Field] = Constants.unsampledValue
+            }
+        }
+    }
+
+    private enum Constants {
+        static let sampledValue = "1"
+        static let unsampledValue = "0"
+        static let b3Separator = "-"
+    }
```

Comment:
Not sure this is necessary, especially since it's not leveraged in `OpenTelemetryHTTPHeadersReader`.

Reply:
Ah right. I had different plan but forgot about it. I moved these to header.